### ISSUE TITLE
Source Dynamodb: aws sdk capes the listTables command to 100, resulting in only showing the first 100 tables in airbyte.

### ISSUE THREAD

**Issue by sebas-500** [airbytehq/airbyte#30437]
### Connector Name

source-dynamodb

### Connector Version

0.1.2

### What step the error happened?

Configuring a new connector

### Revelant information

When trying to setup a DynamoDB source connector for an aws account with more than 100 dynamo tables, only the first 100 are showing in airbyte ui. 

The `listTables()` method that you can find in the following file `airbyte-integrations/connectors/source-dynamodb/src/main/java/io/airbyte/integrations/source/dynamodb/DynamodbOperations.java` should be updated.

```
  public List<String> listTables() {
    return dynamoDbClient.listTables()
        // filter on table status?
        .tableNames();
  }
```

By default the listTables() method of the aws sdk lists the first hundred tables. To list every table in the account you can follow steps defined `https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/example_dynamodb_ListTables_section.html`

### Contribute

- [X] Yes, I want to contribute

**Comment by sebas-500** [comment]
The listTables method should be implemented as follow

`  public List<String> listTables() {
    ListTablesRequest request = ListTablesRequest.builder().build();
    List<String> tableNamesList = new ArrayList<>();

    do {
      ListTablesResponse response = dynamoDbClient.listTables(request);
      tableNamesList.addAll(response.tableNames());

      request = request.toBuilder().exclusiveStartTableName(response.lastEvaluatedTableName()).build();
    } while (request.exclusiveStartTableName() != null);

    return tableNamesList;
  }`

**Comment by sebas-500** [comment]
Hi, I was wondering if there's any update on the status of this issue.

**Comment by davidmartinsutc** [comment]
I had the exact same issue. Instead of make a local fix, I pushed a PR: https://github.com/airbytehq/airbyte/pull/31913

**Comment by mpiccoli-eiwa** [comment]
Hey guys, hope you're all doing well.
I ran into the same issue, and was wondering if we could merge the PR with the fix. Also, if there's anything I can do to help, just let me know.
Thanks!.

**Comment by marcosmarxm** [comment]
Closed by #31935
